protocols/kad/src/behaviour.rs

```diff
@@ -1210,9 +1217,13 @@ where
     fn connection_updated(
         &mut self,
         peer: PeerId,
-        address: Option<Multiaddr>,
+        mut address: Option<Multiaddr>,
         new_status: NodeStatus,
     ) {
+        if let Some(addr) = address {
+            address = addr.with_p2p(peer).ok();
+        }
```

Comment:
Wouldn't it make most sense to do the `with_p2p` sanitizing at the borders of the crate only? Given that `connection_updated` is a private method, I would expect a caller to do the sanitization.

Do others share the same intuition?

Reply:
After your comment I reconsidered why I had done this in `connection_updated` and `on_address_change`. And after re-running my tests, I found out it was no longer necessary since the case is now directly handled in `protocols.rs`. So I removed it :wink: